src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -224,13 +224,15 @@ object LoadVCF {
       val hd = parseHeader(reader, getHeaderLines(hConf, file))
       val hd1 = header1Bc.value
 
-      hd1.sampleIds.iterator.zip(hd.sampleIds.iterator)
-        .zipWithIndex.dropWhile { case ((s1, s2), i) => s1 == s2 }.toArray.headOption match {
-        case Some(((s1, s2), i)) => fatal(
-          s"""invalid sample ids: expected sample ids to be identical for all inputs. Found different sample ids at position $i.
-           |    ${ files(0) }: $s1
-           |    $file: $s2""".stripMargin)
-        case None =>
+      if (!hd1.sampleIds.iterator.sameElements(hd.sampleIds.iterator)) {
```

Comment:
I don't think you need this if-statement. Also, this check could be written more concisely with `.foreach` rather `.dropWhile` and I think you can accomplish the same thing as `sameElements` without iterating through the data twice.

Reply:
I had the .sameElements check in there on the theory that it's probably faster and consumes less memory than the other check, so if there's no error to be thrown then it won't bother creating the value-index tuples. If that's not true, though, I'll take it out. I'll fix the other thing shortly.